internal/datastore/common/gc.go

```diff
@@ -149,14 +149,20 @@ var MaxGCInterval = 60 * time.Minute
 // StartGarbageCollector loops forever until the context is canceled and
 // performs garbage collection on the provided interval.
 func StartGarbageCollector(ctx context.Context, collectable GarbageCollectableDatastore, interval, window, timeout time.Duration) error {
-	return startGarbageCollectorWithMaxElapsedTime(ctx, collectable, interval, window, 0, timeout, gcFailureCounter)
+	return runPeriodicallyWithBackoff(ctx, func() error {
```

Comment:
And this is where we're actually using it - we thunk the garbage collection logic.